http-support/src/main/java/org/eclipse/sw360/antenna/http/utils/FailedRequestException.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.http.utils;
+
+import java.io.IOException;
+
+/**
+ * <p>
+ * A special exception class to report errors caused by requests that have an
+ * unexpected (typically non-success) status code.
+ * </p>
+ * <p>
+ * This exception class stores the response status code, so that it is
+ * available for further investigation. A cause is not available because the
+ * request completed normally. In order to identify the request that failed, a
+ * tag can be provided that is added to the exception message. This is useful
+ * if a future is constructed from the results of multiple requests.
+ * </p>
+ */
+public class FailedRequestException extends IOException {
```

Comment:
Do we use this really, as I wrote I do not like new exception types but you added a semantic here, which I haven't found caught and used anywhere

Reply:
In this case, the new exception class brings additional value by storing the status code received from the server. This is used for different purposes in the SW360 client library. For instance, it is checked whether the status code is 401, indicating that the access token has expired. When using a standard exception for this purpose, we would somehow have to parse the status code out of the exception message - which I would really like to avoid.

Reply:
Ok